include/swift/AST/DiagnosticsSema.def

```diff
@@ -4632,6 +4632,10 @@ NOTE(objc_optional_requirement_swift_rename,none,
      "rename %select{method|initializer|property|subscript}0 to match "
      "requirement %1", (bool, DeclName))
 
+ERROR(witness_objc_default_implementation, none, 
+      "default implementations cannot be used to satisfy requirements of "
+      "'@objc' protocols, such as %0 in '@objc' protocol %1", (DeclName, DeclName))
```

Comment:
Sorry, this has totally slipped from my radar. If we use `such as` here maybe a better wording would be singular e.g. `cannot use default implementation of <member-kind> <member-name> to satisfy requirement of '@obj' protocol <protocol-name>`?

Reply:
Hmm that's similar to what I suggested earlier, but @amomchilov [pointed out](https://github.com/apple/swift/pull/33891#discussion_r487592660) that it doesn't make it clear enough that the idea of using default implementations is not allowed in general for `@objc` protocols and instead it makes it look like that there is something wrong with the particular requirement, so I tweaked the wording a bit.

I have another idea - we can take what you suggested above and also include this bit: `default implementations cannot be used to satisfy requirements of '@objc' protocols`. It makes the diagnostic text a bit longer, though.

Reply:
Maybe add that as a note?

Reply:
I guess the problem is a duality of it - you need to express that _this_ particular requirement can't be satisfied and at the same mention that action is also invalid in general. Any ideas, @hborla?

Reply:
I think it would be more consistent and actionable to say `ConformingClass does not conform to ObjCProtocol` with a fix-it to add the stubs, and emit a tailored note that says something like `default implementations cannot be used to satisfy requirements of '@objc' protocols`. What do you think?

Reply:
Yeah, that sounds good to me!

Reply:
Great idea, I have done that now. Let me know what you think!